test/e2e_node/container_lifecycle_test.go

```diff
@@ -51,6 +51,7 @@ func prefixedName(namePrefix string, name string) string {
 
 var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", func() {
 	f := framework.NewDefaultFramework("containers-lifecycle-test")
+	cleanupPods(f)
```

Comment:
how will this work for non-serial tests? There is no guarantee that a pod will not be created after we called this function. here: https://github.com/kubernetes/kubernetes/pull/123468/files we actually had serial tests

Reply:
`cleanupPods` is misnamed. Let me rename it to reflect exactly what it does namely `addAfterEachForCleaningUpPods`

All we are doing in this PR is to ensure that when the test finishes, we explicitly cleanup pods so when the serial tests are started there are no straggling pods.

Reply:
One more thing to call out @SergeyKanzhelev - if you see the https://github.com/kubernetes/kubernetes/pull/123468/files#diff-6f3fbea74d4f9dfbdb8ee8b6f9415a1a266750ff6ebc7946ae496f18e1687fbcR82-R83 that was added in that PR you referenced, that's the cause of all the flakes :)

that test assumes that no pods are there on the node which is not true. So in this PR we are trying to clean up pods as soon as the respective tests finish in the hope that we can reduce the flakes from that memory manager test

cc @ffromani